askama_parser/src/lib.rs

```diff
@@ -249,11 +248,75 @@ fn bool_lit(i: &str) -> ParseResult<'_> {
     alt((keyword("false"), keyword("true")))(i)
 }
 
+/// Underscore separateed digits of the given base, unless `must_start_with_digit` this may start
+/// with an underscore.
+fn separated_digits(radix: u32, must_start_with_digit: bool) -> impl Fn(&str) -> ParseResult<'_> {
+    move |i| {
+        recognize(tuple((
+            |i| {
+                if must_start_with_digit {
+                    Ok((i, 0))
+                } else {
+                    many0_count(char('_'))(i)
+                }
+            },
+            satisfy(|ch| ch.is_digit(radix)),
+            many0_count(satisfy(|ch| ch == '_' || ch.is_digit(radix))),
+        )))(i)
+    }
+}
+
 fn num_lit(i: &str) -> ParseResult<'_> {
+    let integer_suffix_tags = |i| {
```

Comment:
Please remove the `_tags` suffixes from this (and `float_suffix_tags`).

Reply:
done